uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-notification-badge/components/tasks-notification-badge.component.js

```diff
@@ -0,0 +1,47 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'tasks-notification-badge/modules/components'
+], function (_, components) {
+  components.component('tasksNotificationBadge', {
+    templateUrl: ['settings', function (settings) {
+      return settings.pathTpl + 'components/tasks-notification-badge.html';
+    }],
+    controllerAs: 'tasksNotificationBadge',
+    controller: TasksNotificationBadgeController
+  });
+
+  TasksNotificationBadgeController.$inject = ['$log', '$q', 'Session'];
+
+  function TasksNotificationBadgeController ($log, $q, Session) {
+    $log.debug('Component: tasks-notification-badge');
+
+    var vm = this;
+    var taskFilters = { apiName: 'Task', params: { status_id: {'!=': 'Completed'} } };
+    var documentFilters = { apiName: 'Document', params: { status_id: 'awaiting upload' } };
+
+    vm.refreshCountEventName = 'TasksBadge:: Update Count';
+
+    (function init () {
+      $q.all([
+        getAssigneeUserId()
+      ]).then(function () {
+        vm.filters = [taskFilters, documentFilters];
+      });
+    })();
+
+    /**
+     * Get the logged in contact id and save it as assignee
+     *
+     * @returns {Promise}
+     */
+    function getAssigneeUserId () {
+      return Session.get()
+        .then(function (session) {
+          taskFilters.params.assignee_contact_id = session.contactId;
```

Comment:
are you really getting the user session id or are you instead appending the user id to the filters?

I suggest the init be refactored like this instead:

```js
(function init () {
  getAssigneeUserId()
    .then(function (assigneeContactId) {
      taskFilters.params.assignee_contact_id = assigneeContactId;
      documentFilters.params.assignee_contact_id = assigneeContactId;
      vm.filters = [taskFilters, documentFilters];
    });
})();

function getAssigneeUserId () {
  return Session.get()
    .then(function (session) {
      return session.contactId;
    });
}
```

Reply:
I decided to use `Session.get` directly, without a separate function.